src/coreclr/gc/gc.cpp

```diff
@@ -23578,8 +24312,677 @@ void gc_heap::equalize_promoted_bytes()
+                heap_segment* tail_ro_region = generation_tail_ro_region (gen);
+                heap_segment* tail_region = generation_tail_region (gen);
+
+                for (heap_segment* region = start_region; region != nullptr; region = heap_segment_next(region))
+                {
+                    set_heap_for_contained_basic_regions (region, hpd);
+                }
+                if (tail_ro_region != nullptr)
+                {
+                    // the first r/w region is the one after tail_ro_region
+                    heap_segment* start_rw_region = heap_segment_next (tail_ro_region);
+
+                    heap_segment* hpd_tail_ro_region = generation_tail_ro_region (hpd_gen);
+                    if (hpd_tail_ro_region != nullptr)
+                    {
+                        // insert the list of r/o regions between the r/o and the r/w regions already present
+                        heap_segment_next (tail_ro_region) = heap_segment_next (hpd_tail_ro_region);
+                        heap_segment_next (hpd_tail_ro_region) = start_region;
+                    }
+                    else
+                    {
+                        // put the list of r/o regions before the r/w regions present
+                        heap_segment_next (tail_ro_region) = generation_start_segment (hpd_gen);
+                        generation_start_segment (hpd_gen) = start_region;
+                    }
+                    generation_tail_ro_region (hpd_gen) = tail_ro_region;
+
+                    // we took care of our r/o regions, we still have to do the r/w regions
+                    start_region = start_rw_region;
+                }
+                // put the r/w regions at the tail of heap 0
+                heap_segment* hpd_tail_region = generation_tail_region (hpd_gen);
+                heap_segment_next (hpd_tail_region) = start_region;
+                generation_tail_region (hpd_gen) = tail_region;
+
+                generation_start_segment (gen) = nullptr;
+                generation_tail_ro_region (gen) = nullptr;
+                generation_tail_region (gen) = nullptr;
+            }
+        }
+    }
+
+    // transfer the free regions from the heaps going idle
+    for (int i = new_n_heaps; i < n_heaps; i++)
+    {
+        gc_heap* hp = g_heaps[i];
+        int dest_heap_number = i % new_n_heaps;
+        gc_heap* hpd = g_heaps[dest_heap_number];
+
+        for (int kind = 0; kind < count_free_region_kinds; kind++)
+        {
+            hpd->free_regions[kind].transfer_regions(&hp->free_regions[kind]);
+        }
+    }
+    // update number of heaps
+    n_heaps = new_n_heaps;
+    gc_t_join.update_n_threads(new_n_heaps);
+#ifdef BACKGROUND_GC
+    bgc_t_join.update_n_threads(new_n_heaps);
+#endif //BACKGROUND_GC
+
+    // even out the regions over the current number of heaps
+    equalize_promoted_bytes (max_generation);
+
+    total_num_fl_items_moved_stage1 = ~0;
+    total_num_fl_items_stage1 = ~0;
+
```

Comment:
```suggestion
```

Reply:
Sure, I'll remove the declarations and uses.